hazelcast/src/main/java/com/hazelcast/security/impl/function/SecuredFunctions.java

```diff
@@ -287,6 +287,13 @@ public void init(@Nonnull ProcessorSupplier.Context context) {
                 dataSource = newDataSourceFn.apply(context);
             }
 
+            @Override
+            public void close(Throwable error) throws Exception {
+                if (dataSource instanceof AutoCloseable) {
+                    ((AutoCloseable) dataSource).close();
+                }
+            }
```

Comment:
Here closing the dataSource is not an issue, because we don't have API for Source#jdbc which would have  `CommonDataSource` supplier.